budgetportal/tests/test_admin_departments_upload.py

```diff
@@ -0,0 +1,126 @@
+from budgetportal.models import (
+    FinancialYear,
+    Sphere,
+    Government,
+    Department,
+)
+from allauth.account.models import EmailAddress
+from ckanapi import NotFound
+from datetime import datetime
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+from mock import Mock
+from selenium import webdriver
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support.ui import Select
+import os
+import sys
+
+USERNAME = 'testuser'
+EMAIL = 'testuser@domain.com'
+PASSWORD = '12345'
+
+
+class AdminDepartmentUploadTest(StaticLiveServerTestCase):
+    def setUp(self):
+        year = FinancialYear.objects.create(slug="2030-31")
+        # spheres
+        self.national_sphere = Sphere.objects.create(financial_year=year, name='National')
+        self.provincial_sphere = Sphere.objects.create(
+            financial_year=year, name='Provincial')
+        # governments
+        self.fake_national_government = Government.objects.create(
+            sphere=self.national_sphere, name='South Africa')
+        self.fake_provincial_government = Government.objects.create(
+            sphere=self.provincial_sphere,
+            name='Free State'
+        )
+
+        user = User.objects.create_user(
+            username=USERNAME,
+            password=PASSWORD,
+            is_staff=True,
+            is_superuser=True,
+            is_active=True,
+        )
+        EmailAddress.objects.create(
+            user=user,
+            email=EMAIL,
+            verified=True,
+        )
+
+        self.selenium = webdriver.PhantomJS()
+
+        self.path = os.path.dirname(__file__)
+
+        super(AdminDepartmentUploadTest, self).setUp()
+
+    def tearDown(self):
+        self.selenium.quit()
+
+        # https://stackoverflow.com/questions/14991244/how-do-i-capture-a-screenshot-if-my-nosetests-fail
+        if sys.exc_info()[0]:  # Returns the info of exception being handled
+            now = datetime.now().strftime('%Y-%m-%d_%H-%M-%S-%f')
+            self.selenium.get_screenshot_as_file('%s.png' % now)
```

Comment:
This screenshot code fails for me, with the WebDriver code unable to connect to the server to send the screenshot command. I think it only actually works before `selenium.quit()`, not after.

(If it's working sometimes, it could be due to a race condition between the quit and screenshot commands?)

Reply:
Should be fixed in 4f8ccd2.